pyiron_nodes/atomistic/engine/vasp.py

```diff
@@ -0,0 +1,263 @@
+    stdout: str
+    stderr: str
+    return_code: int
+
+
+@Workflow.wrap.as_function_node("output")
+def run_job(
+    command: str,
+    workdir: str = os.getcwd(),
+    environment: Optional[Dict[str, str]] = None,
+    arguments: Optional[List[str]] = None,
+) -> ShellOutput:
+    if environment is None:
+        environment = {}
+    if arguments is None:
+        arguments = []
+    print(f"run_job is in {os.getcwd()}")
+    environ = dict(os.environ)
+    environ.update({k: str(v) for k, v in environment.items()})
+    proc = subprocess.run(
+        [command, *map(str, arguments)],
+        capture_output=True,
+        cwd=workdir,
+        encoding="utf8",
+        env=environ,
+        shell=True,
+    )
+    print(f"Running job in directory: {workdir}")
+    output = ShellOutput()
+    output.stdout = proc.stdout
+    output.stderr = proc.stderr
+    print(proc.stdout, proc.stderr)
+    output.return_code = proc.returncode
+    return output
+
+
+@Workflow.wrap.as_function_node("output_dict")
+def parse_VaspOutput(workdir: str) -> Dict:
+    print(f"workdir of parse: {workdir}")
+    return pvo(workdir)
+
+
+@Workflow.wrap.as_function_node("convergence")
+def check_convergence(
+    workdir: str,
+    filename_vasprun: str = "vasprun.xml",
+    filename_vasplog: str = "vasp.log",
+    backup_vasplog: str = "error.out",
+) -> bool:
+    converged = False
+    line_converged = (
+        "reached required accuracy - stopping structural energy minimisation"
+    )
+
+    try:
+        vr = Vasprun(filename=os.path.join(workdir, filename_vasprun))
+        converged = vr.converged
+    except:
+        try:
+            converged = is_line_in_file(
+                filepath=os.path.join(workdir, filename_vasplog),
+                line=line_converged,
+                exact_match=False,
+            )
+        except:
+            try:
+                converged = is_line_in_file(
+                    filepath=os.path.join(workdir, backup_vasplog),
+                    line=line_converged,
+                    exact_match=False,
+                )
+            except:
+                pass
+
+    return converged
+
+
+@Workflow.wrap.as_macro_node("vasp_output", "convergence_status")
+def vasp_job(
+    self,
+    workdir: str,
+    vasp_input,
+    command: str = "module load vasp; module load intel/19.1.0 impi/2019.6; unset I_MPI_HYDRA_BOOTSTRAP; unset I_MPI_PMI_LIBRARY; mpiexec -n 1 vasp_std",
+):
+    self.working_dir = create_WorkingDirectory(workdir=workdir)
+    self.vaspwriter = write_VaspInputSet(workdir=workdir, vasp_input=vasp_input)
+    self.job = run_job(command=command, workdir=workdir)
+    self.vasp_output = parse_VaspOutput(workdir=workdir)
+    self.convergence_status = check_convergence(workdir=workdir)
+
+    (
+        self.working_dir
+        >> self.vaspwriter
+        >> self.job
+        >> self.vasp_output
+        >> self.convergence_status
+    )
+    self.starting_nodes = [self.working_dir]
```

Comment:
I still have a nagging feeling that needing to specify the execution flow is a code smell, and it means something can be done with the data IO of the constituents. With that said, I took another look and VASP is just so file-centric that I find your current IO very sensible and have no alternatives to suggest. No need to change anything here, but let's tuck away in our heads the idea that maybe this is smelly and maybe there's some clever way to get around it that we just don't see yet.